optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/SequenceImpl.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.common.experimental.impl;
+
+import java.util.NavigableSet;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.optaplanner.examples.common.experimental.api.Sequence;
+
+public class SequenceImpl<ValueType_, DifferenceType_ extends Comparable<DifferenceType_>>
+        implements Sequence<ValueType_, DifferenceType_> {
+    private NavigableSet<ValueType_> consecutiveItemsSet;
+    private ValueType_ firstItem;
+    private ValueType_ lastItem;
+    private final ConsecutiveSetTree<ValueType_, ?, DifferenceType_> sourceTree;
+
+    protected SequenceImpl(ConsecutiveSetTree<ValueType_, ?, DifferenceType_> sourceTree, ValueType_ item) {
```

Comment:
If the type is hidden, I don't see the need to hide the methods. 

I generally try to hide the type, and make public methods public. This is an API within an API - those who have access to the type should be able to use the type, except for methods you want to keep hidden even from them.